clang/docs/SYCLFERFCIntegrationHeaderFooter.md

```diff
@@ -0,0 +1,141 @@
+# Introduction
+SYCL source files are logically (and usually physically) compiled multiple times; code that runs on the "host" (e.g., a CPU) is compiled by a host compiler and code that runs on a "device" (e.g., a GPU) is compiled by a device compiler. The SYCL language is designed such that it is not necessary that the host compiler be SYCL-aware; an ISO C++ conforming compiler suffices. However, some SYCL features such as kernel invocation and support for
+[specialization constants](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_specialization_constants)
+require coordination between the host and device compilers. Integration headers and footers produced by a device compiler are used to communicate the information needed during host compilation to support these features.
+
+# Integration Header
+An integration header is a source file produced during device compilation that is intended to be pre-included during host compilation of the same source file.
+
+Consider the following partial code:
+
+```
+  sycl::accessor<char, 1, sycl::access::mode::read> acc1;
+  int i = 13;
+  sycl::sampler smplr;
+  struct S {
+    char c;
+    int i;
+  } test_s;
+  test_s.c = 14;
+  kernel_single_task<class first_kernel>([=]() {
+    if (i == 13 && test_s.c == 14) {
+
+      acc1.use();
+      smplr.use();
+    }
+  });
+```
```

Comment:
I think it would look more professional to show real SYCL code here, rather than this SYCL-like pseudo code.  I would also avoid using `sycl::sampler` in the example because this is our own extension on top of SYCL 2020.  I don't think it adds anything to your example, and it might raise questions about supporting syntax that is beyond the scope of the standard.  Putting this together, the example might look like this:

```
int main() {
  sycl::queue q;
  sycl::buffer<char> b{sycl::range{1024}};

  q.submit([&]{sycl::handler &cgh) {
    sycl::accessor acc{b, cgh};
    int i;
    struct S {
      char c;
      int i;
    } test_s;
    test_s.c = 14;

    cgh.single_task[=] {
      if (i == 13 && test_s.c == 14) {
        acc[0] = 'a';
      }
    });
  });
}
```